compiler/rustc_ast/src/tokenstream.rs

```diff
@@ -611,7 +611,7 @@ impl Cursor {
     }
 
     pub fn look_ahead(&self, n: usize) -> Option<&TokenTree> {
-        self.stream.0[self.index..].get(n).map(|(tree, _)| tree)
+        self.stream.0.get(self.index..).and_then(|elem| elem.get(n)).map(|(tree, _)| tree)
```

Comment:
why?

Reply:
Hehehehe, couldn't resist. Sorry :P

Usually, I tend to avoid things that will panic at runtime but this indexing is perfectly safe. Let me remove this modification.

Reply:
Thanks! Normally, I would agree, but in the compiler I would be wary of such changes. Often, the index/length of arrays means something. If they are not what we expect, weird stuff will happen that could be hard to debug. This seems to be especially true in the type system, too. Removing the panic would actually delay the error to happen somewhere else, where it could be more confusing.